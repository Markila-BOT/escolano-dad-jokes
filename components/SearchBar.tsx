import React, { Dispatch, FC, SetStateAction } from "react";

type SearchBarTypes = {
  query: string;
  setQuery: Dispatch<SetStateAction<string>>;
};

const SearchBar: FC<SearchBarTypes> = ({ query, setQuery }) => {
  function handleOnSearch({ currentTarget = {} }) {
    const { value } = currentTarget as { value: string };
    setQuery(value);
  }

  return (
    <div
      className="flex items-center justify-center"
      data-testid="search-bar-main"
    >
      <div className="flex border-2 rounded">
        <input
          name="search"
          type="search"
          autoComplete="off"
          id="search"
          className="px-4 py-2 w-80"
          placeholder="Search keywords on joke"
          value={query}
          onChange={handleOnSearch}
        />
        <div className="flex items-center justify-center px-4">
          <svg
            className="w-6 h-6 text-gray-600"
            fill="currentColor"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
          >
            <path d="M16.32 14.9l5.39 5.4a1 1 0 0 1-1.42 1.4l-5.38-5.38a8 8 0 1 1 1.41-1.41zM10 16a6 6 0 1 0 0-12 6 6 0 0 0 0 12z" />
          </svg>
        </div>
      </div>
    </div>
  );
};

export default SearchBar;
