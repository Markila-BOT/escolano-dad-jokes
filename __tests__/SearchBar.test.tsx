import { render, screen } from "@testing-library/react";
import SearchBar from "../components/SearchBar";

describe("SearchBar", () => {
  it("should renders the SearchBar component", () => {
    const { container } = render(<SearchBar query="" setQuery={() => {}} />);
    expect(screen.getByTestId("search-bar-main")).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
