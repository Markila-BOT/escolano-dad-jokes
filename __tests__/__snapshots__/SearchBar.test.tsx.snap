// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should renders the SearchBar component 1`] = `
<div>
  <div
    class="flex items-center justify-center"
    data-testid="search-bar-main"
  >
    <div
      class="flex border-2 rounded"
    >
      <input
        autocomplete="off"
        class="px-4 py-2 w-80"
        id="search"
        name="search"
        placeholder="Search keywords on joke"
        type="search"
        value=""
      />
      <div
        class="flex items-center justify-center px-4"
      >
        <svg
          class="w-6 h-6 text-gray-600"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.32 14.9l5.39 5.4a1 1 0 0 1-1.42 1.4l-5.38-5.38a8 8 0 1 1 1.41-1.41zM10 16a6 6 0 1 0 0-12 6 6 0 0 0 0 12z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
